craft_parts/plugins/conda_plugin.py

```diff
@@ -0,0 +1,123 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2022 Canonical Ltd.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License version 3 as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""The conda plugin."""
+
+import logging
+from typing import Any, Dict, List, Optional, Set, cast
+
+from . import validator
+from .base import Plugin, PluginModel, extract_plugin_properties
+from .properties import PluginProperties
+
+logger = logging.getLogger(__name__)
+
+
+class CondaPluginProperties(PluginProperties, PluginModel):
+    """The part properties used by the conda plugin."""
+
+    # part properties required by the plugin
+    conda_packages: Optional[List[str]] = None
+    conda_python_version: Optional[str] = None
+    conda_install_prefix: str = "/snap/${CRAFT_PROJECT_NAME}/current"
```

Comment:
is the project name not available through options?

Reply:
this is probably a snapcraft plugin if this is the default

Reply:
As discussed with @sergiusens, craft-parts plugins can't assume the resulting artifact will be a snap. If this is essential for conda to work, this should be a snapcraft plugin instead.

Reply:
How about this logic in `get_build_commands`?
```
        if options.conda_install_prefix:
            install_prefix = options.conda_install_prefix
        elif self._part_info.application_name == "snapcraft":
            install_prefix = f"/snap/{self._part_info.project_name}/current"
        else:
            install_prefix = str(self._part_info.part_install_dir)

        deploy_command = [
            f"CONDA_TARGET_PREFIX_OVERRIDE={install_prefix}",
```